lib/src/Uncertainty/Algorithm/Sensitivity/HSICEstimatorTargetSensitivity.cxx

```diff
@@ -0,0 +1,222 @@
+//                                               -*- C++ -*-
+/**
+ * @brief HSICEstimatorTargetSensitivity implements the target sensitivity HSIC indices
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/HSICEstimatorTargetSensitivity.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+CLASSNAMEINIT(HSICEstimatorTargetSensitivity)
+
+/* Default */
+HSICEstimatorTargetSensitivity::HSICEstimatorTargetSensitivity()
+  : HSICEstimatorImplementation()
+  , isAlreadyComputedPValuesAsymptotic_(false)
+{
+  // Nothing to do
+}
+
+/* Constructor */
+HSICEstimatorTargetSensitivity::HSICEstimatorTargetSensitivity(
+  const CovarianceModelCollection & covarianceList
+  , const Sample & X
+  , const Sample & Y
+  , const HSICStat & estimatorType
+  , const Function & filterFunction )
+  : HSICEstimatorImplementation(covarianceList, X, Y, estimatorType)
+  , isAlreadyComputedPValuesAsymptotic_(false)
+{
+  filterFunction_ =  filterFunction;
+  /* apply filter */
+  outputSample_ = filterFunction_(outputSample_);
+}
+
+/* Virtual constructor */
+HSICEstimatorTargetSensitivity* HSICEstimatorTargetSensitivity::clone() const
+{
+  return new HSICEstimatorTargetSensitivity(*this);
+}
+
+/* Compute the weight matrix from a sample */
+SquareMatrix HSICEstimatorTargetSensitivity::computeWeightMatrix(const Sample&) const
```

Comment:
I don't understand why this method takes a Sample as an argument as it is unused. You don't need to have `computeWeightMatrix` at the interface of the base class as it is not used outside of the class. You don't even have to take the same name for this method among the classes derived from HSICEstimatorImplementation. You implemented an actual weighted version in `HSICEstimatorImplementation` only to use it in `HSICEstimatorConditionalSensitivity`. Move this method to HSICEstimatorConditionalSensitivity and remove it from HSICEstimatorImplementation.

Reply:
The problem is that `HSICEstimatorImplementation::computePValuesPermutation` and `HSICEstimatorImplementation::computeIndices` both call `computeWeightMatrix`.

Reply:
On second thought, since the default weight function is constant with value 1, we actually don't need to overload `computeWeightMatrix` in `HSICEstimatorGlobalSensitivity` and `HSICEstimatorTargetSensitivity`: indeed, `HSICEstimatorImplementation::computeWeightMatrix` will by default return an identity matrix when no weight function is provided.

Reply:
So, at the CT, we agreed not to change the  `HSICEstimatorGlobalSensitivity` and `HSICEstimatorTargetSensitivity` versions of `computeWeightMatrix` but to move the `HSICEstimatorImplementation` version to `HSICEstimatorConditionalSensitivity`.